src/basic/parse-util.c

```diff
@@ -228,15 +228,14 @@ int parse_size(const char *t, uint64_t base, uint64_t *size) {
                         return -ERANGE;
 
                 r += tmp;
-                if ((unsigned long long) (uint64_t) r != r)
-                        return -ERANGE;
-
                 p = e + strlen(table[i].suffix);
-
                 start_pos = i + 1;
 
         } while (*p);
 
+        if (r > UINT64_MAX)
+                return -ERANGE;
```

Comment:
hmm, on systems where "unsigned long long" is 64 bit isn't this going to result in a compiler warning complaining about a comparison that can never be true? Isn't this just shifting warnings between lgtm and gcc here?

Reply:
I don't think so, but if we aren't 100% certain, it would be better to leave everything as is. Another option would be to add `// lgtm[cpp/comparison-of-identical-expressions]` to suppress the alerts if they annoy anybody. I personally can live with them.

Reply:
I have no idea about that. The reason I submit this PR is just I prefer that, and optionally, it works as a workaround for the false-positive errors in lgtm.

Reply:
So, please say no if you or people do not like that. Then I will drop them.

Reply:
I'm sorry. I didn't mean to say that I don't like it. It's just that I don't know how to answer the question @poettering asked with 100% certainty. When I'm in doubt, my first kind of jerk-knee reaction is not to change anything. Sometimes it helps; sometimes it doesn't :-)